db/tests/dependents/test_dependents.py

```diff
@@ -0,0 +1,148 @@
+        ]
+    )
+    assert all(
+        [
+            r['level'] == 4
+            for r in checkouts_dependents
+        ]
+    )
+
+
+def test_response_format(engine, library_tables_oids):
+    publishers_dependents_graph = get_dependents_graph(library_tables_oids['Publishers'], engine)
+
+    dependent_expected_attrs = ['obj', 'parent_obj', 'level']
+    obj_expected_attrs = ['objid', 'type']
+    assert all(
+        [
+            all(attr in dependent for attr in dependent_expected_attrs)
+            for dependent in publishers_dependents_graph
+        ]
+    )
+    assert all(
+        [
+            all(attr in dependent['obj'] for attr in obj_expected_attrs)
+            for dependent in publishers_dependents_graph
+        ]
+    )
+    assert all(
+        [
+            all(attr in dependent['parent_obj'] for attr in obj_expected_attrs)
+            for dependent in publishers_dependents_graph
+        ]
+    )
+
+
+# TODO: add other types when they are added as dependents
+def test_specific_object_types(engine, library_tables_oids, library_db_tables):
+    items_oid = library_tables_oids['Items']
+    items_dependents_graph = get_dependents_graph(items_oid, engine)
+    items_dependents_oids = _get_object_dependents_oids(items_dependents_graph, items_oid)
+
+    items_constraint_oids = [
+        get_constraint_oid_by_name_and_table_oid(constraint.name, items_oid, engine)
+        for constraint in library_db_tables['Items'].constraints]
+
+    checkouts_oid = library_tables_oids['Checkouts']
+    checkouts_items_fk = [c for c in library_db_tables['Checkouts'].foreign_key_constraints if 'Item' in c][0]
+    checkouts_items_fk_oid = get_constraint_oid_by_name_and_table_oid(checkouts_items_fk.name, checkouts_oid, engine)
+
+    assert sorted(items_dependents_oids) == sorted(items_constraint_oids + [checkouts_oid] + [checkouts_items_fk_oid])
+
+
+# if a table contains a foreign key referencing itself, it shouldn't be treated as a dependent
+def test_self_reference(engine, library_tables_oids):
+    with engine.begin() as conn:
+        conn.execute(text('ALTER TABLE "Publishers" ADD COLUMN "Parent Publisher" integer'))
+        conn.execute(text('ALTER TABLE "Publishers" ADD CONSTRAINT "Publishers_Publisher_fkey" FOREIGN KEY ("Parent Publisher") REFERENCES "Publishers" (id)'))
+
+    publishers_oid = library_tables_oids['Publishers']
+    publishers_dependents_graph = get_dependents_graph(publishers_oid, engine)
+
+    publishers_dependents_oids = _get_object_dependents_oids(publishers_dependents_graph, publishers_oid)
+    assert publishers_oid not in publishers_dependents_oids
+
+
+# if two tables depend on each other, we should return dependence only for the topmost object in the graph
+# excluding the possibility of circulal reference
+def test_circular_reference(engine, library_tables_oids):
+    with engine.begin() as conn:
+        conn.execute(text('ALTER TABLE "Publishers" ADD COLUMN "Top Publication" integer'))
```

Comment:
We should add a self-referential field to the dataset instead of creating it in the test case. I have added an issue for it here https://github.com/centerofci/mathesar-data-playground/issues/5. Don't worry about it making the change now, Just add a to-do, so that we can refactor it later.

Reply:
Test cases are throwing up an error because it is not able to determine the schema, instead of using a raw SQL statement it can be replaced with an equivalent function `db.columns.operations.create.create_column`